packages/opencensus-instrumentation-grpc/test/test-grpc.ts

```diff
@@ -0,0 +1,493 @@
+    server = startServer(grpcModule, proto);
+    client = createClient(grpcModule, proto);
+  });
+
+  beforeEach(() => {
+    rootSpanVerifier.endedRootSpans = [];
+  });
+
+  after(() => {
+    server.forceShutdown();
+  });
+
+  const requestList: TestRequestResponse[] = [{num: 100}, {num: 50}];
+  const resultSum = {
+    num: requestList.reduce(
+        (sum, x) => {
+          return sum + x.num;
+        },
+        0)
+  };
+  const methodList = [
+    {
+      description: 'unary call',
+      methodName: 'UnaryMethod',
+      method: grpcClient.unaryMethod,
+      request: requestList[0],
+      result: requestList[0]
+    },
+    {
+      description: 'clientStream call',
+      methodName: 'ClientStreamMethod',
+      method: grpcClient.clientStreamMethod,
+      request: requestList,
+      result: resultSum
+    },
+    {
+      description: 'serverStream call',
+      methodName: 'ServerStreamMethod',
+      method: grpcClient.serverStreamMethod,
+      request: resultSum,
+      result: replicate(resultSum)
+    },
+    {
+      description: 'bidiStream call',
+      methodName: 'BidiStreamMethod',
+      method: grpcClient.bidiStreamMethod,
+      request: requestList,
+      result: requestList
+    }
+  ];
+
+  // Compare two arrays using an equal function f
+  // tslint:disable-next-line:no-any
+  const arrayIsEqual = (f: any) => ([x, ...xs]) => ([y, ...ys]): any =>
+      x === undefined && y === undefined ?
+      true :
+      Boolean(f(x)(y)) && arrayIsEqual(f)(xs)(ys);
+
+  // Return true if two requests has the same num value
+  const requestEqual = (x: TestRequestResponse) => (y: TestRequestResponse) =>
+      x.num !== undefined && x.num === y.num;
+
+  // Check if its equal requests or array of requests
+  const checkEqual = (x: TestRequestResponse|TestRequestResponse[]) =>
+      (y: TestRequestResponse|TestRequestResponse[]) =>
+          (x instanceof Array) && (y instanceof Array) ?
+      (arrayIsEqual(requestEqual)(x)(y)) :
+      !(x instanceof Array) && !(y instanceof Array) ? (requestEqual(x)(y)) :
+                                                       false;
+
+  function assertSpan(
+      span: types.Span, spanName: string, type: string,
+      status: grpcModule.status) {
+    assert.strictEqual(span.name, spanName);
+    assert.strictEqual(span.type, type);
+    assert.strictEqual(span.status, plugin.traceStatus(status));
+
+    assert.strictEqual(span.attributes[GrpcPlugin.ATTRIBUTE_GRPC_KIND], type);
+    assert.strictEqual(
+        span.attributes[GrpcPlugin.ATTRIBUTE_GRPC_STATUS_CODE], `${status}`);
+    assert.ok(span.attributes[GrpcPlugin.ATTRIBUTE_GRPC_METHOD]);
+    if (status !== grpcModule.status.OK) {
+      assert.ok(span.attributes[GrpcPlugin.ATTRIBUTE_GRPC_ERROR_NAME]);
+      assert.ok(span.attributes[GrpcPlugin.ATTRIBUTE_GRPC_ERROR_MESSAGE]);
+    }
+  }
+
+  function assertPropagation(sourceSpan: types.Span, targetSpan: types.Span) {
```

Comment:
Can you comment on what this function is asserting? It is hard to tell at a glance which is the server or client.

Reply:
Done!